cartridge/twophase.lua

```diff
@@ -507,29 +507,9 @@ local function _clusterwide(patch)
         }):lock()
     end
 
-    local clusterwide_config_new = clusterwide_config_old:copy()
-    for k, v in pairs(patch) do
-        if patch[k] ~= nil and patch[k .. '.yml'] ~= nil then
-            local err = PatchClusterwideError:new(
-                'Ambiguous sections %q and %q',
-                k, k .. '.yml'
-            )
-            log.error('%s', err)
-            return nil, err
-        end
-        if v == nil then
-            clusterwide_config_new:set_plaintext(k, v)
-            clusterwide_config_new:set_plaintext(k .. '.yml', patch[k .. '.yml'])
-        elseif type(v) == 'string' then
-            clusterwide_config_new:set_plaintext(k, v)
-        else
-            if not string.endswith(k, '.yml') then
-                clusterwide_config_new:set_plaintext(k, box.NULL)
-                k = k .. '.yml'
-            end
-
-            clusterwide_config_new:set_plaintext(k, yaml.encode(v))
-        end
+    local clusterwide_config_new, err = clusterwide_config_old:patch_copy(patch)
```

Comment:
I suggest keeping it split as before:

```lua
local clusterwide_config_new = clusterwide_config_old:copy()
local _, err = clusterwide_config_new:patch(patch)
```

But +1 for moving it to clusterwide module, I like it too.

Reply:
> I suggest keeping it split as before:

In case if we consider clusterwide_config as immutable object correct way is to create copy for each object modification.

Reply:
I would prefer to treat it as an immutable object as well.

Reply:
Ok